### PR TITLE
Lift functions from Lazy.ByteString and System.Directory (closes #55)

### LINKED ISSUE #55: Replace reexports with lifted versions of functions
Functions specialized to `IO` should be generalized to either `MonadIO` or `MonadUnliftIO`. Applies at least to the modules `RIO.ByteString`, `RIO.Directory`, and `RIO.ByteString.Lazy`. Pointed out by @adituv in #53.

### PR DESCRIPTION
**Pull request by roman**

### CODE REVIEW

File: src/RIO/ByteString/Lazy.hs [hunk 1]
```diff
@@ -1,141 +1,211 @@
...
 --
 -- > import qualified RIO.ByteString.Lazy as B.Lazy
+{-# LANGUAGE NoImplicitPrelude #-}
 module RIO.ByteString.Lazy
   (
   -- * The @ByteString@ type
     Data.ByteString.Lazy.ByteString
 
   -- * Introducing and eliminating 'ByteString's
-  , Data.ByteString.Lazy.empty
```
Comment (snoyberg) [inline src/RIO/ByteString/Lazy.hs]: I'm not a fan of including unrelated changes like renamings in a PR like this. To be honest, I preferred the original naming in the module. Would you be able to undo that renaming?

**Comment by roman** [comment]
@adituv I went ahead and rebased my changes on top of yours, cheers.

**Comment by adituv** [comment]
Ahh I was going to squash mine and tidy up the commits first but oh well!  So long as it's all in there
